From b13566a0bad7b6f14ea117b60fa090f3c6dd5931 Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@armbian.com>
Date: Sat, 13 Jul 2024 15:11:08 -0400
Subject: [PATCH] bananapi m2s fixup

HDMI doesn't come up until handing off to the kernel, at
which point it only displays in green and magenta. Removing
the default *-u-boot.dtsi and adding one specific to each variant
corrects the issue.

Signed-off-by: Patrick Yavitz <pyavitz@armbian.com>
---
 ...boot.dtsi => meson-g12b-a311d-bananapi-m2s-u-boot.dtsi} | 0
 arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi     | 7 +++++++
 2 files changed, 7 insertions(+)
 rename arch/arm/dts/{meson-g12b-bananapi-u-boot.dtsi => meson-g12b-a311d-bananapi-m2s-u-boot.dtsi} (100%)
 create mode 100644 arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi

diff --git a/arch/arm/dts/meson-g12b-bananapi-u-boot.dtsi b/arch/arm/dts/meson-g12b-a311d-bananapi-m2s-u-boot.dtsi
similarity index 100%
rename from arch/arm/dts/meson-g12b-bananapi-u-boot.dtsi
rename to arch/arm/dts/meson-g12b-a311d-bananapi-m2s-u-boot.dtsi
diff --git a/arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi b/arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi
new file mode 100644
index 0000000000..236f2468dc
--- /dev/null
+++ b/arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2019 BayLibre, SAS.
+ * Author: Neil Armstrong <narmstrong@baylibre.com>
+ */
+
+#include "meson-g12-common-u-boot.dtsi"
-- 
2.39.2

